### DISCUSSION TITLE
Migrating problem

### DISCUSSION THREAD

**Discussion by nico-hl** [immich-app/immich#6064] [Q&A]
Hey,

i hope someone can help me out.
I have now decided to set up a new Unraid server (smaller and more refined) 😂

I have successfully migrated all Docker containers, but now I want to optimize a few things with Immich, and unfortunately, I'm currently stuck.

I have created a dump as described here [[insert link].](https://immich.app/docs/administration/backup-and-restore/)

I set up Immich with the Docker Compose plugin. However, I have noticed that there is also Immich as a proper app in the store. I find this a bit more convenient as I feel it provides better security and more control over the individual Docker containers (Immich, Redis & Postgres).

So, I installed all the Docker containers, restored the dump, and started Immich. So far, the log shows no errors 😁

Now, the page with the photos loads, and there's quite a bit going on in the log:

`[Nest] 176  - 12/30/2023, 6:39:18 AM   ERROR [Error: ENOENT: no such file or directory, access 'upload/thumbs/be7e3455-449c-485a-bb16-05acd2126d57/39f9e3ba-2f9b-4980-8a6f-5cc41e846a19.jpeg'] Unable to send file: Error
[Nest] 176  - 12/30/2023, 6:39:18 AM    WARN [ExpressAdapter] Content-Type doesn't match Reply body, you might need a custom ExceptionFilter for non-JSON responses
[Nest] 176  - 12/30/2023, 6:39:18 AM   ERROR [ExceptionsHandler] ENOENT: no such file or directory, access 'upload/thumbs/be7e3455-449c-485a-bb16-05acd2126d57/39f9e3ba-2f9b-4980-8a6f-5cc41e846a19.jpeg'
Error: ENOENT: no such file or directory, access 'upload/thumbs/be7e3455-449c-485a-bb16-05acd2126d57/39f9e3ba-2f9b-4980-8a6f-5cc41e846a19.jpeg'
[Nest] 176  - 12/30/2023, 6:39:18 AM   ERROR [Error: ENOENT: no such file or directory, access 'upload/thumbs/be7e3455-449c-485a-bb16-05acd2126d57/9444d170-dfdd-4006-9b9c-68508a358d9b.jpeg'] Unable to send file: Error
[Nest] 176  - 12/30/2023, 6:39:18 AM    WARN [ExpressAdapter] Content-Type doesn't match Reply body, you might need a custom ExceptionFilter for non-JSON responses
[Nest] 176  - 12/30/2023, 6:39:18 AM   ERROR [ExceptionsHandler] ENOENT: no such file or directory, access 'upload/thumbs/be7e3455-449c-485a-bb16-05acd2126d57/9444d170-dfdd-4006-9b9c-68508a358d9b.jpeg'
Error: ENOENT: no such file or directory, access 'upload/thumbs/be7e3455-449c-485a-bb16-05acd2126d57/9444d170-dfdd-4006-9b9c-68508a358d9b.jpeg'
[Nest] 176  - 12/30/2023, 6:39:18 AM   ERROR [Error: ENOENT: no such file or directory, access 'upload/thumbs/be7e3455-449c-485a-bb16-05acd2126d57/68b8cb25-8fc6-4967-9937-1132dd6fca57.jpeg'] Unable to send file: Error
[Nest] 176  - 12/30/2023, 6:39:18 AM    WARN [ExpressAdapter] Content-Type doesn't match Reply body, you might need a custom ExceptionFilter for non-JSON responses
[Nest] 176  - 12/30/2023, 6:39:18 AM   ERROR [ExceptionsHandler] ENOENT: no such file or directory, access 'upload/thumbs/be7e3455-449c-485a-bb16-05acd2126d57/68b8cb25-8fc6-4967-9937-1132dd6fca57.jpeg'
Error: ENOENT: no such file or directory, access 'upload/thumbs/be7e3455-449c-485a-bb16-05acd2126d57/68b8cb25-8fc6-4967-9937-1132dd6fca57.jpeg'
[Nest] 176  - 12/30/2023, 6:39:18 AM   ERROR [Error: ENOENT: no such file or directory, access 'upload/thumbs/be7e3455-449c-485a-bb16-05acd2126d57/2e/26/2e2640f6-c3f0-42d2-99be-0e63a46a8f15.webp'] Unable to send file: Error
[Nest] 176  - 12/30/2023, 6:39:18 AM    WARN [ExpressAdapter] Content-Type doesn't match Reply body, you might need a custom ExceptionFilter for non-JSON responses
[Nest] 176  - 12/30/2023, 6:39:18 AM   ERROR [ExceptionsHandler] ENOENT: no such file or directory, access 'upload/thumbs/be7e3455-449c-485a-bb16-05acd2126d57/2e/26/2e2640f6-c3f0-42d2-99be-0e63a46a8f15.webp'
Error: ENOENT: no such file or directory, access 'upload/thumbs/be7e3455-449c-485a-bb16-05acd2126d57/2e/26/2e2640f6-c3f0-42d2-99be-0e63a46a8f15.webp'`

The paths are adjusted in Docker:

Photos: /mnt/user/Immich/
Upload: /mnt/user/Immich/

In the Immich folder, the Upload directory is located, and I moved the Thumbs directory into it because it was previously in the Immich root directory.

I see now in the Immich Admin Section under Repair he tells me following path isnt there: 

/app/immich/server/upload ...

Is this a path from the postgres backup from the old server/docker ? where can i change it ? 

It would be great if someone has a tip for me :)

nico

**Comment by ProxiBlue** [comment]
Hello,

I ran into same issue whilst migrating immich to a HomeAssistant OS run server (as aposed to a previously standalone docker run)
I had dumpe dmy db, and have my assets on usb drive.

see my investigation and details here: https://github.com/alexbelgium/hassio-addons/discussions/1467#discussioncomment-11703476

The short:

i placed a symlink on immich wb instance here: /app/immich/server

```ln -s /share/immich ./upload```

/share/immich is teh folder defined as upload_location

maybe something similar will work for you....
